### ISSUE TITLE
[11.0] stock: ERROR: null value in column "route_id" violates not-null constraint

### ISSUE THREAD

**Issue by davidtranhp** [OCA/OpenUpgrade#1417]
### Impacted versions:
11.0
### Steps to reproduce:
```
./odoo-bin -d tvtma10 -u all --stop-after-init
```
### Current behavior:
```
2018-06-29 14:30:54,343 4079 INFO tvtma10 OpenUpgrade: stock: post-migration script called with version 10.0.1.1
2018-06-29 14:30:54,362 4079 DEBUG tvtma10 OpenUpgrade: Running 
        UPDATE product_template
        SET responsible_id = create_uid
        
2018-06-29 14:30:54,362 4079 DEBUG tvtma10 OpenUpgrade: 192 rows affected
2018-06-29 14:30:54,366 4079 ERROR tvtma10 odoo.sql_db: bad query: b'\n            INSERT INTO procurement_rule\n            SELECT t.id, t.create_date, t.sequence, t.write_uid, t.write_date, t.active, t.create_uid, t.name, t.company_id, t.action, t.group_id, t.group_propagation_option, t.partner_address_id, t.warehouse_id, t.propagate, t.procure_method, t.location_id, t.location_src_id, t.route_sequence_moved0, t.picking_type_id, t.delay, t.route_id, t.propagate_warehouse_id, t.invoice_state, t.route_sequence\n            FROM openupgrade_legacy_11_0_procurement_rule t, stock_location_route slr\n            '
ERROR: null value in column "route_id" violates not-null constraint
DETAIL:  Failing row contains (6, 2017-04-09 16:21:29.655631, 20, 1, 2017-04-09 16:47:07.735791, t, 1, Your Company:  Manufacture, null, manufacture, null, propagate, null, 1, f, make_to_stock, 12, null, null, 7, 0, null, null, null, 0).

2018-06-29 14:30:54,366 4079 ERROR tvtma10 OpenUpgrade: stock: error in migration script stock/migrations/11.0.1.1/post-migration.py: null value in column "route_id" violates not-null constraint
DETAIL:  Failing row contains (6, 2017-04-09 16:21:29.655631, 20, 1, 2017-04-09 16:47:07.735791, t, 1, Your Company:  Manufacture, null, manufacture, null, propagate, null, 1, f, make_to_stock, 12, null, null, 7, 0, null, null, null, 0).

2018-06-29 14:30:54,366 4079 ERROR tvtma10 OpenUpgrade: null value in column "route_id" violates not-null constraint
DETAIL:  Failing row contains (6, 2017-04-09 16:21:29.655631, 20, 1, 2017-04-09 16:47:07.735791, t, 1, Your Company:  Manufacture, null, manufacture, null, propagate, null, 1, f, make_to_stock, 12, null, null, 7, 0, null, null, null, 0).
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 1427, in wrapped_function
    if use_env2 else cr, version)
  File "/home/tuan/git/openupgrade11/addons/stock/migrations/11.0.1.1/post-migration.py", line 154, in migrate
    create_specific_procurement_rules_from_globals(env)
  File "/home/tuan/git/openupgrade11/addons/stock/migrations/11.0.1.1/post-migration.py", line 42, in create_specific_procurement_rules_from_globals
    openupgrade.get_legacy_name(table)
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 923, in logged_query
    cr.execute(query, args)
  File "/home/tuan/git/openupgrade11/odoo/sql_db.py", line 155, in wrapper
    return f(self, *args, **kwargs)
  File "/home/tuan/git/openupgrade11/odoo/sql_db.py", line 232, in execute
    res = self._obj.execute(query, params)
psycopg2.IntegrityError: null value in column "route_id" violates not-null constraint
DETAIL:  Failing row contains (6, 2017-04-09 16:21:29.655631, 20, 1, 2017-04-09 16:47:07.735791, t, 1, Your Company:  Manufacture, null, manufacture, null, propagate, null, 1, f, make_to_stock, 12, null, null, 7, 0, null, null, null, 0).

2018-06-29 14:30:54,367 4079 ERROR tvtma10 odoo.modules.loading: Error executing post migration script for module stock
`-> to_location_warehouse
: null value in column "route_id" violates not-null constraint
DETAIL:  Failing row contains (6, 2017-04-09 16:21:29.655631, 20, 1, 2017-04-09 16:47:07.735791, t, 1, Your Company:  Manufacture, null, manufacture, null, propagate, null, 1, f, make_to_stock, 12, null, null, 7, 0, null, null, null, 0).

2018-06-29 14:30:54,373 4079 WARNING tvtma10 odoo.modules.loading: Transient module states were reset
2018-06-29 14:30:54,376 4079 ERROR tvtma10 odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/home/tuan/git/openupgrade11/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/tuan/git/openupgrade11/odoo/modules/loading.py", line 442, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check, upg_registry)
  File "/home/tuan/git/openupgrade11/odoo/modules/loading.py", line 328, in load_marked_modules
    upg_registry=upg_registry,
  File "/home/tuan/git/openupgrade11/odoo/modules/loading.py", line 208, in load_module_graph
    migrations.migrate_module(package, 'post')
  File "/home/tuan/git/openupgrade11/odoo/modules/migration.py", line 174, in migrate_module
    migrate(self.cr, installed_version)
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 1427, in wrapped_function
    if use_env2 else cr, version)
  File "/home/tuan/git/openupgrade11/addons/stock/migrations/11.0.1.1/post-migration.py", line 154, in migrate
    create_specific_procurement_rules_from_globals(env)
  File "/home/tuan/git/openupgrade11/addons/stock/migrations/11.0.1.1/post-migration.py", line 42, in create_specific_procurement_rules_from_globals
    openupgrade.get_legacy_name(table)
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 923, in logged_query
    cr.execute(query, args)
  File "/home/tuan/git/openupgrade11/odoo/sql_db.py", line 155, in wrapper
    return f(self, *args, **kwargs)
  File "/home/tuan/git/openupgrade11/odoo/sql_db.py", line 232, in execute
    res = self._obj.execute(query, params)
psycopg2.IntegrityError: null value in column "route_id" violates not-null constraint
DETAIL:  Failing row contains (6, 2017-04-09 16:21:29.655631, 20, 1, 2017-04-09 16:47:07.735791, t, 1, Your Company:  Manufacture, null, manufacture, null, propagate, null, 1, f, make_to_stock, 12, null, null, 7, 0, null, null, null, 0).

2018-06-29 14:30:54,377 4079 CRITICAL tvtma10 odoo.service.server: Failed to initialize database `tvtma10`.
Traceback (most recent call last):
  File "/home/tuan/git/openupgrade11/odoo/service/server.py", line 925, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/home/tuan/git/openupgrade11/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/tuan/git/openupgrade11/odoo/modules/loading.py", line 442, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check, upg_registry)
  File "/home/tuan/git/openupgrade11/odoo/modules/loading.py", line 328, in load_marked_modules
    upg_registry=upg_registry,
  File "/home/tuan/git/openupgrade11/odoo/modules/loading.py", line 208, in load_module_graph
    migrations.migrate_module(package, 'post')
  File "/home/tuan/git/openupgrade11/odoo/modules/migration.py", line 174, in migrate_module
    migrate(self.cr, installed_version)
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 1427, in wrapped_function
    if use_env2 else cr, version)
  File "/home/tuan/git/openupgrade11/addons/stock/migrations/11.0.1.1/post-migration.py", line 154, in migrate
    create_specific_procurement_rules_from_globals(env)
  File "/home/tuan/git/openupgrade11/addons/stock/migrations/11.0.1.1/post-migration.py", line 42, in create_specific_procurement_rules_from_globals
    openupgrade.get_legacy_name(table)
  File "/usr/local/lib/python3.6/dist-packages/openupgradelib/openupgrade.py", line 923, in logged_query
    cr.execute(query, args)
  File "/home/tuan/git/openupgrade11/odoo/sql_db.py", line 155, in wrapper
    return f(self, *args, **kwargs)
  File "/home/tuan/git/openupgrade11/odoo/sql_db.py", line 232, in execute
    res = self._obj.execute(query, params)
psycopg2.IntegrityError: null value in column "route_id" violates not-null constraint
DETAIL:  Failing row contains (6, 2017-04-09 16:21:29.655631, 20, 1, 2017-04-09 16:47:07.735791, t, 1, Your Company:  Manufacture, null, manufacture, null, propagate, null, 1, f, make_to_stock, 12, null, null, 7, 0, null, null, null, 0).

```

**Comment by davidtranhp** [comment]
I found that my database has a rule named Manufacture which is a legacy of the previously-uninstalled MRP.
I solved the problem by deleting that rule by executing an SQL query. Is there a better way?

**Comment by pedrobaeza** [comment]
Indeed that's something not fixable at migration stage, as it's a previous inconsistency (as that field is required), so you should fix it previously.

**Comment by pedrobaeza** [comment]
It turns out that actually there's an error on the migration script. I didn't detect it in first tests because I didn't have any global rule on my tests DBs. I have fixed it in #1419, specifically in https://github.com/OCA/OpenUpgrade/pull/1419/files#diff-fe7bbc3631adade92da541853a857311R65
